Additional Params/Fields

grahamlyus opened the pull request:
Great script!

I've added support for additional params/fields via a configuration option additionalParamsMap - it works the same way as the UTM params, but is user configurable.

Also a minor tweak to the build script, as I wasn't seeing the minified script updating.

grahamlyus committed:
add a build script and make uglify depend on coffee for building, so the minified js actually gets built.

grahamlyus committed:
add additionalParamsMap to enable additional params/fields to be added to the form.

medius commented:
Thanks @grahamlyus for the PR. Looks great.

How about also bumping the version in https://github.com/medius/utm_form/blob/master/package.json#L3, say 1.0.1, and checking in the new generated files? It will certainly help with CDN, browser caches and such without other manual changes to file names.

grahamlyus committed:
Bump version and Update README.md.
Also added a clean task and combined coffee and uglify tasks, as I was still not getting both files reliably on build.

grahamlyus commented:
Done. I was still having issues with gulp not building both scripts, and combining the coffee and uglify tasks seems to fix this (I've never used gulp before). Also I added a clean task to remove the previous versions and updated the README for you - be aware that I bumped the versions in the README that you reference in your CloudFront distribution.

medius commented:
Thanks for the fixes. I really appreciate it. Looks good to merge.
